src/com/amazon/ion/benchmark/WriteRandomIonValues.java

```diff
@@ -604,6 +653,23 @@ public static BigDecimal constructDecimal(List<Integer> expValRange, List<Intege
         return coefficient.scaleByPowerOfTen(exp);
     }
 
+    /**
+     * Construct the decimal which is conformed with the constraints provided in ISL.
+     * @param precision represents the minimum/maximum range indicating the number of digits in the unscaled value of a decimal. The minimum precision must be greater than or equal to 1.
+     * @param scale represents the minimum/maximum range indicating the number of digits to the right of the decimal point. The minimum scale must be greater than or equal to 0.
+     * @return the constructed decimal.
+     */
+    public static BigDecimal constructDecimalFromSchema(int precision, int scale) {
+        Random random = new Random();
+        StringBuilder rs = new StringBuilder();
+        for (int digit = 0; digit < precision; digit++) {
+            rs.append(random.nextInt(9) + 1);
```

Comment:
Should this be `rs.append(random.nextInt(10));`?

Reply:
> Should this be `rs.append(random.nextInt(10));`?

This is used for generating random integer from 1 to 9 which is following this formula random.nextInt((max - min) + 1) + min

Reply:
Right, but don't you want to generate random integers between 0 and 9 (except for the first digit, which should be between 1 and 9)?

Reply:
> Right, but don't you want to generate random integers between 0 and 9 (except for the first digit, which should be between 1 and 9)?

Right! Thank you for pointing out. The first digit should be random int between 1 to 9, and the rest of the digits should be random int between 0 to 9 . This will be updated in the following commit.